File: test/gcd.rb
#!/usr/bin/env ruby

# GCD. We assume positive numbers
def gcd(a, b)
  # Make: a <= b
  if a > b
    a, b = [b, a]
  end

  return nil if a <= 0

  if a == 1 or b-a == 0
    return a
  end
  return gcd(b-a, a)
end

a, b = ARGV[0..1].map {|arg| arg.to_i}
puts "The GCD of %d and %d is %d" % [a, b, gcd(a, b)]

File: test/gcd.pl
#!/usr/bin/env perl
use strict;
use warnings;

# GCD. We assume positive numbers
sub gcd($$);
sub gcd($$) 
{ 
    my ($a, $b) = @_;
    # Make: a <= b
    ($a, $b) = ($b, $a) if ($a > $b);

    return undef if $a <= 0;
    return $a if ($a == 1) or ($b-$a == 0);
    return gcd($b-$a, $a);
}

die sprintf "Need two integer arguments, got %d", scalar(@ARGV) unless 
    @ARGV == 2;
my ($a, $b) = @ARGV[0,1];
printf "The GCD of %d and %d is %d\n", $a, $b, gcd($a, $b);
